docs/tutorials/typescript/nextjs/index.md

```diff
@@ -1,290 +1,626 @@
+Workflows orchestrate [Activities](https://docs.temporal.io/activities), which is how you interact with the outside world in Temporal.
+You use Activities to make API requests, access the filesystem, or perform other non-deterministic operations.
+
+The Workflow you'll create in this tutorial will have a single Activity, `purchase`.
+
+Create the file `temporal/src/activities.ts`:
+
+```command
+touch temporal/src/activities.ts
+```
 
-Inside of `/temporal/src/activities.ts` we'll write a simple Activity function to start with:
+Inside of `temporal/src/activities.ts`, add the following code to define the `purchase` function:
 
+<!--SNIPSTART typescript-next-oneclick-activities -->
+[temporal/src/activities.ts](https://github.com/temporalio/nextjs-temporal-one-click-template/blob/next-v2/temporal/src/activities.ts)
 ```ts
-// /temporal/src/activities.ts
 import { Context } from '@temporalio/activity';
 
 export async function purchase(id: string): Promise<string> {
   console.log(`Purchased ${id}!`);
   return Context.current().info.activityId;
 }
 ```
+<!--SNIPEND-->
+
+In a real application, this function would interact with a payment API and attempt to make a payment.
+For this tutorial, the function prints a message to the console and returns the ID of the Activity.
 
-Activities are the only way to interact with the outside world in Temporal (e.g. making API requests, or accessing the filesystem).
-See the [Activities docs](https://docs.temporal.io/dev-guide/typescript/foundations//#develop-activities) for more info.
+Now define the `order` Workflow that calls this Activity.
+Create the file `temporal/src/workflows.ts`:
+
+```command
+touch temporal/src/workflows.ts
+```
 
-Inside of `/temporal/src/workflows.ts` we'll write a Workflow function that calls this Activity:
+Add the following code to `temporal/src/workflows.ts` to define the Workflow:
 
+<!--SNIPSTART typescript-next-oneclick-workflows -->
+[temporal/src/workflows.ts](https://github.com/temporalio/nextjs-temporal-one-click-template/blob/next-v2/temporal/src/workflows.ts)
 ```ts
-// /temporal/src/workflows.ts
 import { proxyActivities, sleep } from '@temporalio/workflow';
-import type * as activities from './activities'; // purely for type safety
+import type * as activities from './activities';
 
 const { purchase } = proxyActivities<typeof activities>({
   startToCloseTimeout: '1 minute',
 });
 
 export async function OneClickBuy(id: string): Promise<string> {
   const result = await purchase(id); // calling the activity
-  await sleep('10 seconds'); // demo use of timer
+  await sleep('10 seconds'); // sleep to simulate a longer response.
   console.log(`Activity ID: ${result} executed!`);
+  return result;
 }
 ```
+<!--SNIPEND-->
+
+Workflows must be deterministic, and you perform non-deterministic work in Activities.
+The TypeScript SDK bundles Workflow code and runs it inside a [deterministic sandbox](https://docs.temporal.io/typescript/determinism).
```

Comment:
This link does not exist. How about this:

https://docs.temporal.io/develop/typescript/core-application#workflow-logic-requirements